src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.SslCtx.cs

```diff
@@ -69,7 +83,107 @@ protected override bool ReleaseHandle()
         {
             Interop.Ssl.SslCtxDestroy(handle);
             SetHandle(IntPtr.Zero);
+            if (_gch.IsAllocated)
+            {
+                //Interop.Ssl.SslCtxSetData(this, (IntPtr)_gch);
```

Comment:
If this line of code was at all necessary it'd presumably need to be before SslCtxDestroy(handle).

```suggestion
```

Reply:
after some more testing I realized that we can get called after the handle is closed and released. (OpenSSL keeps it's own references) I refactor the cleanup quite a bit.